pkg/netflow/flowaggregator/aggregator.go

```diff
@@ -119,6 +124,56 @@ func (agg *FlowAggregator) sendFlows(flows []*common.Flow) {
 	}
 }
 
+func (agg *FlowAggregator) sendExporterMetadata(flows []*common.Flow, flushTime time.Time) {
+	// exporterMap structure: map[NAMESPACE]map[EXPORTER_ID]metadata.NetflowExporter
+	exporterMap := make(map[string]map[string]metadata.NetflowExporter)
+
+	// orderedExporterIDs is used to build predictable metadata payload (consistent batches and orders)
+	// orderedExporterIDs structure: map[NAMESPACE]EXPORTER_ID
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**❓ question:** I'm missing the part where we order the exporter IP to get a consistent order. Can you help me understand?‏

Reply:
I needed the exporters to be ordered mainly for testing, but I think it's also a good thing for troubleshooting.

The comment `map[NAMESPACE]EXPORTER_ID` is actually not correct, should be `map[NAMESPACE][]EXPORTER_ID`.

The list of `[]EXPORTER_ID` is ordered in sense that, when a exporter in the `flows` is seen, it will be added to the list.

About `map[NAMESPACE]` part, this part is not ordered. Let me see if I can also order this part.


The `orderedExporterIDs` are used later in the code when building the payload.

Reply:
My bad, actually we don't need to order namespaces since they are sent in separate payloads.

Reply:
Back to your initial question the `orderedExporterIDs` contains the list of IDs in the order of arrival.

Then later here we us those ordered IDs:

https://github.com/DataDog/datadog-agent/blob/4a1cfd9c85b71ecff8003ac8c98f65b7c457fc24/pkg/netflow/flowaggregator/aggregator.go#L156-L160

Reply:
I see, so the order is consistent but they don't have to be sorted. Makes sense 👍